assess/templates/assess/macros/application_overviews_table_all.html

```diff
@@ -92,8 +92,9 @@ <h2 class="govuk-error-summary__title">
         <input type="checkbox" name="selected_assessments" aria-labelledby="project-name-{{ loop.index }}" value="{{ overview.application_id }}" {% if overview.application_id in display_config["selected_assessments"] %}checked{% endif %} />
       </td>
       {% endif %}
-      <td class="govuk-table__cell">{{ overview.short_id[-6:] }}</td>
-      <td class="govuk-table__cell"><a class="govuk-link" data-qa="project_name" id="project-name-{{ loop.index }}"
+      {% set application_reference=overview.short_id[-6:] %}
+      <td class="govuk-table__cell">{{ application_reference }}</td>
+      <td class="govuk-table__cell"><a class="govuk-link" data-testid="{{application_reference}}" data-qa="project_name" id="project-name-{{ loop.index }}"
```

Comment:
Very minor, could this describe the thing that's being set? My guess if reading the test consuming would be that there's a uniquely generated ID for this test rather than an attribute meaningfully belonging to the thing being enumerated. Maybe something like `data-application-reference=`

Reply:
That's a fair point. I was going with `test-id` as there's a [playwright function](https://playwright.dev/python/docs/locators#locate-by-test-id) for retrieving by that tag, but I guess we could use a different tag and then use x-path to search?

Reply:
Hmm yeh reading through the docs you linked does seem to encourage that consistent label, if it's idiomatic for the testing framework then I guess you only need to learn it once and you'll understand. Happy with either! (But guessing the test code will read cleaner with your original code)